src/libraries/analytics/analyticsEventTracking.ts

```diff
@@ -32,18 +22,43 @@ function getAnalyticsInstance(userId?: string) {
 
   try {
     if (userId) {
-      SEGMENT_INSTANCE.userInfo.set({
-        anonymousId: userId,
-        userId,
-        traits: {name: profile?.username, id: userId}
-      });
-
-      MIXPANEL_INSTANCE.identify(userId);
-      MIXPANEL_INSTANCE.getPeople().set('$name', profile?.username);
+      if (SEGMENT_INSTANCE) {
+        SEGMENT_INSTANCE.userInfo.set({
+          anonymousId: userId,
+          userId,
+          traits: {name: profile?.username, id: userId}
+        });
+      }
+
+      if (MIXPANEL_INSTANCE) {
+        MIXPANEL_INSTANCE.identify(userId);
+        MIXPANEL_INSTANCE.getPeople().set('$name', profile?.username);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code is setting user information for both Segment and Mixpanel instances. However, there's no error handling in case these operations fail. Consider wrapping each operation in a try-catch block to handle potential errors.

```diff
-      if (SEGMENT_INSTANCE) {
-        SEGMENT_INSTANCE.userInfo.set({
-          anonymousId: userId,
-          userId,
-          traits: {name: profile?.username, id: userId}
-        });
-      }
-
-      if (MIXPANEL_INSTANCE) {
-        MIXPANEL_INSTANCE.identify(userId);
-        MIXPANEL_INSTANCE.getPeople().set('$name', profile?.username);
-      }

+      try {
+        if (SEGMENT_INSTANCE) {
+          SEGMENT_INSTANCE.userInfo.set({
+            anonymousId: userId,
+            userId,
+            traits: {name: profile?.username, id: userId}
+          });
+        }
+      } catch (e) {
+        console.error('Error setting Segment user info', e);
+      }
+
+      try {
+        if (MIXPANEL_INSTANCE) {
+          MIXPANEL_INSTANCE.identify(userId);
+          MIXPANEL_INSTANCE.getPeople().set('$name', profile?.username);
+        }
+      } catch (e) {
+        console.error('Error setting Mixpanel user info', e);
+      }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->